### PR TITLE
SWATCH-1715: Create scaffolding for swatch-producer-azure

### PR DESCRIPTION
**Pull request by kahowell**
Jira issue: [SWATCH-1715](https://issues.redhat.com/browse/SWATCH-1715)

Description
===========

Create swatch-producer-azure with stubbed out behaviors.

Note I also refactored kafka seeking logic and the PSK into shared libraries.

Testing
=======

Setup
-----

Deploy the service:

```shell
./gradlew swatch-producer-azure:quarkusDev
```

Verification
------------

Try the following curl command to simulate some seemingly applicable usage

```shell
curl -X 'POST' \
  'http://localhost:8000/api/swatch-producer-azure/internal/azure/billable_usage' \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "account_number": "account123",
  "org_id": "org123",
  "id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "billing_provider": "azure",
  "billing_account_id": "string",
  "snapshot_date": "2022-03-10T12:15:50-04:00",
  "product_id": "rosa",
  "uom": "Cores",
  "value": 0,
  "billing_factor": 0,
  "vendor_product_code": "string"
}'
```

View the logs and see that a message is produced and then consumed for the billable usage.

### CODE REVIEW

File: bin/build-images.sh [hunk 1]
```diff
@@ -68,7 +69,10 @@ for p in "${projects[@]}"; do
     "swatch-contracts")
       podman build . -f swatch-contracts/src/main/docker/Dockerfile.jvm -t quay.io/$quay_user/swatch-contracts:$tag --label "git-commit=${commit}" --ulimit nofile=2048:2048
       ;;
-    *) echo "Please use values from the set \"rhsm\", \"conduit\", \"swatch-producer-aws\", \"swatch-contracts\"";;
+    "swatch-producer-azure")
```
Comment (lindseyburnett) [inline bin/build-images.sh:72]: Corresponding "push and clean" entry needed around line 100

```
    "swatch-producer-azure")
      push_and_clean "quay.io/$quay_user/swatch-producer-azure:$tag"
      ;;
```

**Review by lindseyburnett — CHANGES_REQUESTED** [review]

**Review by lindseyburnett — APPROVED** [review]

File: bin/build-images.sh [hunk 2]
```diff
@@ -42,6 +42,7 @@ if [ ${#projects[@]} -eq 0 ]; then
   projects[1]="conduit"
   projects[2]="swatch-producer-aws"
   projects[3]="swatch-contracts"
+  projects[3]="swatch-producer-azure"
```
Comment (Sgitario) [inline bin/build-images.sh:45]:
```suggestion
  projects[4]="swatch-producer-azure"
```

**Review by Sgitario — CHANGES_REQUESTED** [review]
Just a minor thing.

File: bin/build-images.sh (hunk 2 above)
Comment (kahowell) [reply bin/build-images.sh]: Thanks for catching that. Fixed.

**Comment by kahowell** [comment]
> Just a minor thing.
I would also add some @QuarkusTest test with some Azure wiremock instance to settle the basis. But we can add this in a separate pull request as well (I can help with that if you want).

@Sgitario, sure. Let's handle in [SWATCH-1726](https://issues.redhat.com/browse/SWATCH-1726). I'll make a note that you can help with the wiremock instance.

**Pull request merged**